lib/fastlane/plugin/wpmreleasetoolkit/helper/strings_file_validation_helper.rb

```diff
@@ -0,0 +1,129 @@
+module Fastlane
+  module Helper
+    class StringsFileValidationHelper
+      # context can be one of:
+      #   :root, :maybe_comment_start, :in_line_comment, :in_block_comment,
+      #   :maybe_block_comment_end, :in_quoted_key,
+      #   :after_quoted_key_before_eq, :after_quoted_key_and_equal,
+      #   :in_quoted_value, :after_quoted_value
+      State = Struct.new(:context, :buffer, :in_escaped_ctx, :found_key, keyword_init: true)
+
+      TRANSITIONS = {
+        root: {
+          /\s/ => :root,
+          '/' => :maybe_comment_start,
+          '"' => :in_quoted_key
+        },
+        maybe_comment_start: {
+          '/' => :in_line_comment,
+          /\*/ => :in_block_comment
+        },
+        in_line_comment: {
+          "\n" => :root,
+          /./ => :in_line_comment
+        },
+        in_block_comment: {
+          /\*/ => :maybe_block_comment_end,
+          /./m => :in_block_comment
+        },
+        maybe_block_comment_end: {
+          '/' => :root,
+          /./m => :in_block_comment
+        },
+        in_quoted_key: {
+          '"' => lambda do |state, _|
+            state.found_key = state.buffer.string.dup
+            state.buffer.string = ''
+            :after_quoted_key_before_eq
+          end,
+          /./ => lambda do |state, c|
+            state.buffer.write(c)
+            :in_quoted_key
+          end
+        },
+        after_quoted_key_before_eq: {
+          /\s/ => :after_quoted_key_before_eq,
+          '=' => :after_quoted_key_and_eq
+        },
+        after_quoted_key_and_eq: {
+          /\s/ => :after_quoted_key_and_eq,
+          '"' => :in_quoted_value
+        },
+        in_quoted_value: {
+          '"' => :after_quoted_value,
+          /./m => :in_quoted_value
+        },
+        after_quoted_value: {
+          /\s/ => :after_quoted_value,
+          ';' => :root
+        }
+      }.freeze
+
+      # Returns all the keys in the given `.strings` file, and the line(s) at
+      # which they are defined.
+      #
+      # @param [String] file The path to the file to inspect.
+      # @return [<Hash>] A `Hash` with keys the key values read from the
+      #         `.strings` file. The value of each entry is an array with the
+      #         line numbers where the key occurs.
+      def self.get_keys_from_strings(file:)
+        keys_with_lines = {}
+
+        state = State.new(context: :root, buffer: StringIO.new, in_escaped_ctx: false, found_key: nil)
+
+        File.readlines(file).each_with_index do |line, line_no|
+          line.chars.each_with_index do |c, col_no|
+            # Handle escaped characters at a global level. This is more
+            # straightforward than having a `TRANSITIONS` table that account
+            # for it.
+            if state.in_escaped_ctx || c == '\\'
+              # Just because we check for escaped characters at the global
+              # level, it doesn't mean we allow them in every context.
```

Comment:
```suggestion
              # Just because we check for escaped characters at the global level, it doesn't mean we allow them in every context.
```
Without the hard-wrapping, that comment line end up being as long as the `raise "…" unless …` line below anyway, so feels a bit strange to hard-wrap the comment while we don't hard-wrap the code itself